the-way-we-work.md

```diff
@@ -0,0 +1,664 @@
+
+A developer's "role" is therefore the sum of all the responsibilities at their own band and of the bands below.
+
+However, as developers increase in seniority their general focus is expected to shift from individual work to a whole-team approach.
+
+**Responsibility Matrix**
+
+The responsibility matrix below shows pay bands across the top and areas of responsibility down the side. The new responsibilities acquired at each role are therefore set out in a columnar fashion.
+
+|ROLE TITLE|Developer (bands 6 & 7)|Senior Developer (band 8a)|Principal Engineer (band 8b)|Development & Test Manager (band 8c)|
+|---| :--------------- | :--------- | :--------------- | -------------- |
+| **FOCUS** | **SELF** | **SINGLE TEAM** | **MULTIPLE TEAMS** | **ALL TEAMS** |
+|**CODE**| Feature Development | System Design and Documentation | Engineering Quality Framework |                |
+|**LEADERSHIP and AUTONOMY**| 20% Time | SDMT Operational Planning | Strategy Implementation and Lean Improvement Projects | Vision and Strategy |
+|**GROWING THE TEAM**| Code Reviews | Mentoring and Pairing | Coaching | Delegation |
+|**DUTY OF CARE**| Buddy to Apprentice | Line Management and Recruitment of Band 6/7 | Line Management and Recruitment of Band 8a and Contractors | Line Management and Recruitment of Band 8b and Contractors |
+|**FINANCE**|  |  | Payments to Contractors | Overall budget |
+
+
+**Time Allocation**
+
+When deciding what to work on next, there is one simple rule:
+
+> Prioritise those activities delivering the most benefit to others
+
+TODO: Possible alternative:
+
+> "Help others before you help yourself"
+
+This will usually mean a developer working on activities in their own column in favour of activities in preceeding columns.
+
+
+## 3. Actions & Behaviours
+
+Actions & Behaviours are the things we expect developers to do within their areas of responsibility.
+
+We believe that if developers do these things, and if they try to always improve how they do them, then surely excellent outcomes will follow. 
+
+Note that according to the responsibility matrix above, only some of the below actions and behaviours will be appropriate for each role.
+
+TODO: This section should be "WHAT and HOW OFTEN". Move details to Tools & Methods
+
+TODO: Alternative approach: Combine into "Actions & Behaviours, Tools & Methods" or "Behaviours & Methods" etc.
+
+### 3.1 Code
+
+**Feature Development**
+
+This work will be carried out within a delivery team.
+
+You will:
+
+- attend sprint planning at the beginning of each sprint
+- complete at least one Jira ticket per sprint and change its status to "Done"
+- pair with other developers where possible
+- commit code which is deployed to production at least once per sprint
+- attend a retrospective at the end of each sprint
+
+**System Design and Documentation**
+
+This work will be carried out within a delivery team.
+
+You will:
+
+- use your technical expertise to lead the design of the solution
```

Comment:
ensure tech stack you are using is within approved tech stack

Reply:
@snehal-vahule hopefully this can be accomplished when we have all the teams going through the engineering quality assessment and one of the quality checks is Tech radar check

Reply:
I will add in "ensure the system conforms to approved NHSUK architecture patterns" at this point in the document and also add in "check the system against the criteria in the Engineering Quality Framework" later on in section 4.3. (I may merge the whole of section 4 into section 3 over the weekend)